src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,73 @@
+package duke;
+
+import java.util.ArrayList;
+
+public class TaskList {
+    private final ArrayList<Task> tasks;
+
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    public int size(){
+        return tasks.size();
+    }
+
+    public void loadTask(Task task){
+        tasks.add(task);
+    }
+
+    public String saveTask(int taskIndex){
+        Task task = tasks.get(taskIndex);
+        return task.saveFormat();
+    }
+
+    public void addTask(String request) throws Exception {
+        Task newTask = RequestParser.getTask(request);
+        loadTask(newTask);
+        System.out.printf("Got it. I've added this task:\n" +
+                "  %s\nNow you have %d task in the list\n"
+                ,newTask, tasks.size());
+    }
+
+    public void doneTask(String request) {
+        try {
+            int taskIndex = RequestParser.getTaskIndex(request.trim());
+            Task task = tasks.get(taskIndex);
+            if (task.isDone()) {
+                System.out.println("This task is already done!");
+            } else {
+                tasks.get(taskIndex).setDone();
+                System.out.printf("Nice! I've marked this task as done:\n" +
+                        "  %s\n", tasks.get(taskIndex));
+            }
+        } catch (ArrayIndexOutOfBoundsException ex) {
+            throw new ArrayIndexOutOfBoundsException("☹ OOPS!!! The format for 'done' is 'done <task number>'");
```

Comment:
If a line seems too long consider making it into 2 lines instead, applies to other parts of your code as well.